modules/stream-shredder/src/test/scala/com/snowplowanalytics/snowplow/rdbloader/shredder/stream/sinks/ProcessingSpec.scala

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2012-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and
+ * limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.shredder.stream.sinks
+
+import java.nio.file.Path
+import java.time.Instant
+
+import cats.implicits._
+
+import cats.effect.{Blocker, ContextShift, IO, Timer}
+
+import fs2.io.{file => fs2File}
+import fs2.{Stream, text}
+
+import io.circe.Json
+import io.circe.optics.JsonPath._
+import io.circe.parser.{ parse => parseCirce }
+
+import com.snowplowanalytics.iglu.client.Client
+import com.snowplowanalytics.iglu.core.SchemaKey
+
+import com.snowplowanalytics.snowplow.rdbloader.common.LoaderMessage
+import com.snowplowanalytics.snowplow.rdbloader.common.config.ShredderConfig
+import com.snowplowanalytics.snowplow.rdbloader.common.transformation.Transformed
+import com.snowplowanalytics.snowplow.rdbloader.shredder.stream.Processing
+import com.snowplowanalytics.snowplow.rdbloader.shredder.stream.Processing.Windowed
+import com.snowplowanalytics.snowplow.rdbloader.shredder.stream.sources.{Parsed, file => FileSource}
+import com.snowplowanalytics.snowplow.rdbloader.shredder.stream.sinks.generic.Record
+import com.snowplowanalytics.snowplow.rdbloader.generated.BuildInfo
+
+import org.specs2.mutable.Specification
+
+class ProcessingSpec extends Specification {
+  import ProcessingSpec._
+  "transform" should {
+    "shred events correctly" in {
+      val (good, bad) = transformTestEvents(resourcePath = "/processing-spec/1/input/events", format = shredFormat)
+
+      val pathsToCheck = List(
+        Transformed.Path.Shredded.Tabular("com.snowplowanalytics.snowplow", "atomic", 1),
+        Transformed.Path.Shredded.Tabular("com.snowplowanalytics.snowplow", "consent_document", 1),
+        Transformed.Path.Shredded.Tabular("com.optimizely", "state", 1)
+      )
+      val transformedMap = getTransformedMap(good, pathsToCheck, 1, LoaderMessage.Format.TSV)
+
+      val expectedBadEvent = getResourceLines("/processing-spec/1/output/bad").head
+
+      bad.length mustEqual 1
+      bad.head.data.value mustEqual expectedBadEvent
+      good.length mustEqual 46
+      checkPathContents(good, pathsToCheck, transformedMap) must beTrue
```

Comment:
I find it hard to understand what `checkPathContents` is testing.  Maybe because it's late.  But I wonder if there is a way to write it more expressively so it's easier for the reader to understand what you're testing.

Reply:
Sure, I will try to make it more readable. Also, will add some comments 👍